pkg/util/templater/template_functions.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package templater
+
+import (
+	"text/template"
+
+	"github.com/Masterminds/sprig/v3"
+	"k8s.io/kops"
+	kopsapi "k8s.io/kops/pkg/apis/kops"
+	"k8s.io/kops/pkg/apis/kops/util"
+)
+
+// templateFuncsMap returns a map if the template functions for this template
+func (r *Templater) templateFuncsMap(tm *template.Template) template.FuncMap {
+	// grab the template functions from sprig which are pretty awesome
+	funcs := sprig.TxtFuncMap()
+
+	funcs["indent"] = indentContent
+	// @step: as far as i can see there's no native way in sprig in include external snippets of code
+	funcs["include"] = func(name string, context map[string]interface{}) string {
+		content, err := includeSnippet(tm, name, context)
+		if err != nil {
+			panic(err.Error())
+		}
+
+		return content
+	}
+
+	funcs["ChannelRecommendedKubernetesUpgradeVersion"] = func(version string) string {
```

Comment:
These function names are pretty long.

This is essentially the recommended patch version for a given kubernetes version and the function below is the recommended minor+patch version for a given kops version.

How about:
```
ChannelRecommendedKubernetesPatchVersion
ChannelRecommendedKubernetesMinorVersion
```

or even just
```
ChannelKubernetesPatchVersion
ChannelKubernetesMinorVersion
```
?

Reply:
Not the easiest functions to name :)

I mainly based the function names after the primary underlying function. I don't mind long function names; it boils down to what most intuitive for users.